build_dependencies.sh

```diff
@@ -96,30 +96,6 @@ do
   die_if_not_installed ${pkg}
 done
 
-# check if CC is not set or a null string
-if [[ -z "${CC+unset}" || -z "${CC}" ]]; then
-  if type clang &>/dev/null; then
-    export CC="${CC:-$(which clang)}"
-    msg "Using default clang as CC=${CC}"
-  else
-    msg "Using default C compiler"
-  fi
-else
-  msg "Using custom CC=${CC}"
-fi
-
-# check if CXX is not set or a null string
-if [[ -z "${CXX+unset}" || -z "${CXX}" ]]; then
-  if type clang++ &>/dev/null; then
-    export CXX="${CXX:-$(which clang++)}"
-    msg "Using default clang++ as CXX=${CXX}"
-  else
-    msg "Using default C++ compiler"
-  fi
-else
-  msg "Using custom CXX=${CXX}"
-fi
-
```

Comment:
We no longer want to force clang by default because vcpkg takes care of finding the correct compiler. Or the user can always set `CC` and `CXX` (or use their own toolchain)